### PR TITLE
CI pipelines

### PR DESCRIPTION
**Pull request by adriankast**
Adds two CI pipelines:

1. A default test pipeline (runs on push and PRs)
    - It installs all package dependencies via `lerna bootstrap` (thus should work with unpublished packages also)
    - Runs npm test on all packages (via `lerna run test --no-bail --stream`) returning 1 if one of it failed

2. A visual test pipeline (runs only on PRs)
    - Installs dependencies just like the first pipeline
    - Runs npm test on the `web` package
    - Uploads generated test results (screenshots and example-downloads) to GitHub

Therefore testing in most packages was improved in the first place.

### CODE REVIEW

File: packages/playground-gist_backend/test.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/* check if the server starts without error message and then shut it down after 3s */
```
Comment (egekorkan) [inline packages/playground-gist_backend/test.js:1]: Probably later but storing API key in Github and then actually trying to submit an example TD would be more desirable as a test?

**Review by egekorkan — APPROVED** [review]

**Comment by egekorkan** [comment]
Some comments:
- Could you add node 15 tests as well. It will not be an LTS but would be good to have tests for it in advance
- There are merge conflicts due to the new merged PR 
- Where would the screenshots be seen?

File: packages/playground-gist_backend/test.js (hunk 1 above)
Comment (adriankast) [reply packages/playground-gist_backend/test.js:1]: Yes I agree. We can add this later as you say. Most tests could be enhanced in the future.

**Comment by adriankast** [comment]
merge conflicts are resolved ✅

**Pull request merged**